File: C++/0884.cpp
#include <string>
#include <unordered_map>
#include <vector>
#include <sstream>

class Solution {
public:
    std::vector<std::string> uncommonFromSentences(std::string s1, std::string s2) {
        std::unordered_map<std::string, int> word_count;
        std::istringstream stream1(s1);
        std::istringstream stream2(s2);
        std::string word;
        std::vector<std::string> ans;

        while (stream1 >> word) {
            word_count[word]++;
        }
        while (stream2 >> word) {
            word_count[word]++;
        }

        for (auto pair : word_count) {
            if (pair.second == 1) {
                ans.push_back(pair.first);
            }
        }
        return ans;
    }
};

File: Python/0884.py
from collections import Counter
from typing import List

class Solution:
    def uncommonFromSentences(self, s1: str, s2: str) -> List[str]:
        word_count = Counter(s1.split() + s2.split())

        ans = []
        for w in word_count:
            if word_count[w] == 1:
                ans.append(w)

        return ans

# Topic: Hashmap
# Time Complexity: O(n)
# Space Complexity: O(n)